### ISSUE TITLE
getNftsForOwner discrepencies between docs, actual results, and typings

### ISSUE THREAD

**Issue by TheDutchCoder** [alchemyplatform/alchemy-sdk-js#202]
### [REQUIRED] Environment

- Browser version: N/A (Node)
- Alchemy SDK version: 2.2.3

### [REQUIRED] Describe the problem
The docs for `getNft` (renamed `getNftsForOwner`) show that, when metadata is enabled, the response should contain an object with keys `ownedNfts` (this is correct), which is an array of objects with properties like `metadata` and `media`.

When running the examples in the docs this is indeed the result that is returned.

However, when running the sdk (2.2.3), the returned result has the following format:
```js
{
      contract: [Object],
      tokenId: '9089',
      tokenType: 'ERC721',
      title: '',
      description: '',
      timeLastUpdated: '2022-11-22T18:34:44.342Z',
      metadataError: 'Malformed token uri, do not retry',
      rawMetadata: [Object],
      tokenUri: [Object],
      media: [],
      spamInfo: undefined,
      balance: 1
},
```

As you can see, there's no `metadata` key at all. However there is an undocumented `rawMetadata` key.
This property contains `{ metadata: [], attributes: [] }`, which is incorrect, because there's no `metadata` property in the typings for `rawMetadata`:
```ts
export interface NftMetadata extends Record<string, any> {
    /** Name of the NFT asset. */
    name?: string;
    /** A human-readable description of the NFT asset. */
    description?: string;
    /** URL to the NFT asset image. */
    image?: string;
    /**
     * The image URL that appears along the top of the NFT asset page. This tends
     * to be the highest resolution image.
     */
    external_url?: string;
    /** Background color of the NFT item. Usually defined as a 6 character hex string. */
    background_color?: string;
    /** The traits, attributes, and characteristics for the NFT asset. */
    attributes?: Array<Record<string, any>>;
}
```

Long story short: none of these 3 things actually line up properly which makes using the API incredibly difficult.

#### How to reproduce:
Implement the alchemy-sdk in a Node environment, and do a query on Goerli for an owner.
The response is not the same as described in the docs and it _certainly_ isn't the same as the result in the docs tool .

#### Relevant code or sample repro:
https://stackblitz.com/edit/github-1wqbue-s658mx?file=example.js

Just run `npm run example` and you'll see my test wallet being used to retrieve a bunch of NFT's (most are "empty").
As you can see, _none_ of them return `metadata`, _all_ of them return the undocumented `rawMetadata`, _some_ of them return populated `media` (this seems to be the most "reliable" property to check.

**Comment by thebrianchen** [comment]
@TheDutchCoder Thanks for filing an issue with a detailed explanation, and apologies for not getting to this sooner. We're currently working on getting our documentation into a better place, so thanks for flagging this!

To clarify:
- The SDK renames the `metadata` field from the response into `rawMetadata`. The fact that the SDK docs say the field is `metadata` is a bug, and we'll have a fix out for that shortly.
- The `rawMetadata` field will normally contain the raw metadata specified by the NFT itself. In the cases where the NFT does not have raw metadata, or if Alchemy cannot fetch it, the field is automatically populated with `{ metadata: [], attributes: [] }`.

Hope this answers some of your questions!

**Comment by thebrianchen** [comment]
I'm going to mark the issue as closed, but feel free to ask any other questions you may have.
